### PR TITLE
Rewrite Self-Supervised Learning Docs with low-level blocks

### LINKED ISSUE #555: Rewrite Docs with Low-Level Building Blocks: Self-supervised Learning
# Rewrite Docs with Low-Level Building Blocks: Self-supervised Learning

The [self-supervised learning] page still contains mid- and high-level building blocks.

- [x] Change the training code to a plain pytorch trainig loop
- [x] Change the embedding code to manually loop through the images

Related to #493.

### PR DESCRIPTION
**Pull request by guarin**
Closes #555 

* Remove SelfSupervisedEmbedding dependency
* Add SimCLR model code
* Add pytorch training loop
* Add SimCLR pytorch lightning model code
* Add embedding generation code

[self_supervised_learning.pdf](https://github.com/lightly-ai/lightly/files/7579815/self_supervised_learning.pdf)

### CODE REVIEW

File: docs/source/getting_started/lightly_at_a_glance.rst [hunk 1]
```diff
@@ -70,19 +70,30 @@ with the normalized temperature-scaled cross entropy loss and simple stochastic
...
+            super().__init__()
+            self.backbone = backbone
+            self.projection_head = SimCLRProjectionHead(hidden_dim, hidden_dim, out_dim)
+
+        def forward(self, x):
+            h = self.backbone(x).flatten(start_dim=1)
+            z = self.projection_head(h)
+            return z
+
+    model = SimCLR(resnet, hidden_dim=512, out_dim=128)
```
Comment (philippmwirth) [inline docs/source/getting_started/lightly_at_a_glance.rst:92]: nice

File: docs/source/getting_started/lightly_at_a_glance.rst [hunk 2]
```diff
@@ -94,28 +105,55 @@ with the normalized temperature-scaled cross entropy loss and simple stochastic
...
-    embedding_model.train_embedding(gpus=1, max_epochs=10)
 
 Congrats, you just trained your first model using self-supervised learning!
 
-You can also train the model using PyTorch Lightning directly.
+You can of course also use `PyTorch Lightning <https://www.pytorchlightning.ai/>`_ to implement and train your model.
 
 .. code-block:: python
 
+    class SimCLR(pl.LightningModule):
```
Comment (philippmwirth) [inline docs/source/getting_started/lightly_at_a_glance.rst:134]: can you add the `pytorch-lightning` import here?

**Review by philippmwirth — COMMENTED** [review]
almost good

File: docs/source/getting_started/lightly_at_a_glance.rst (hunk 2 above)
Comment (guarin) [reply docs/source/getting_started/lightly_at_a_glance.rst:135]: Done

**Review by philippmwirth — APPROVED** [review]

**Pull request merged**